pybamm/spatial_methods/finite_volume.py

```diff
@@ -135,22 +135,30 @@ def divergence(self, symbol, discretised_symbol, boundary_conditions):
         submesh = self.mesh[symbol.domain]
 
         divergence_matrix = self.divergence_matrix(symbol.domains)
+        coord_sys = submesh.coord_sys
 
         # check coordinate system
-        if submesh.coord_sys in ["cylindrical polar", "spherical polar"]:
+        if coord_sys in ["cylindrical polar", "spherical polar"]:
             second_dim_repeats = self._get_auxiliary_domain_repeats(symbol.domains)
             # create np.array of repeated submesh.edges
             r_edges_numpy = np.kron(np.ones(second_dim_repeats), submesh.edges)
             r_edges = pybamm.Vector(r_edges_numpy)
-            if submesh.coord_sys == "spherical polar":
+            if coord_sys == "spherical polar":
                 out = divergence_matrix @ ((r_edges**2) * discretised_symbol)
-            elif submesh.coord_sys == "cylindrical polar":
+            elif coord_sys == "cylindrical polar":
                 out = divergence_matrix @ (r_edges * discretised_symbol)
-        else:
+        elif coord_sys == "cartesian":
```

Comment:
there are a few other places where there is similar logic that should be updated (do a search for `submesh.coord_sys in ["cylindrical polar", "spherical polar"]`)